types/match.go

```diff
@@ -23,6 +23,8 @@ import (
 )
 
 // Matcher defines a matching rule for the value of a given label.
+//
+// TODO(asquare14): This is now only used for silences and might be replaced by labels.Matchers.
```

Comment:
Strictly speaking, with just this PR in, it's also still used by inhibitions (which you plan to tackle in a separate PR). Perhaps mention it here and update the comment in the next PR.